jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ErrorProcessorTest.java

```diff
@@ -689,6 +702,25 @@ public void process(Request request, Response response, Callback callback)
         assertThat(response, containsString("Server Error"));
     }
 
+    @Test
+    public void testRootReHandlingErrorProcessor() throws Exception
+    {
+        ReHandlingErrorProcessor.ByCode errorProcessor = new ReHandlingErrorProcessor.ByCode(server);
+        errorProcessor.put(400, "/ok/badMessage");
+        server.setErrorProcessor(errorProcessor);
+
+        String rawResponse = connector.getResponse("""
+                GET /no/host HTTP/1.1
+                
+                """);
+
+        HttpTester.Response response = HttpTester.parseResponse(rawResponse);
+
+        assertThat(response.getStatus(), is(200));
+        assertThat(response.getField(HttpHeader.CONTENT_LENGTH).getIntValue(), greaterThan(0));
+        assertThat(response.getContent(), containsString("/ok/badMessage Error 400 : No Host"));
+    }
+
```

Comment:
Please write a test case for a possible infinite loop, where the re-handle generates the same error that it is trying to handle.